### PR TITLE
Another step closer to autoupgrade

### PR DESCRIPTION
**Pull request by crisidev**
* d220365 <bigo@crisidev.org> 2021-01-30 17:45:10 +0000 (HEAD -> close_to_upgrade, origin/close_to_upgrade, main) Spawn a tokio executor with multiple available workers to let the background task contacting github releases to work
* 5fd6398 <bigo@crisidev.org> 2021-01-30 17:44:20 +0000 Use async signal handling compatibile with tokio
* b657b6c <bigo@crisidev.org> 2021-01-30 17:43:37 +0000 Deprecate docker command for the time being
* 4eec842 <bigo@crisidev.org> 2021-01-30 17:43:11 +0000 Fix github release optional values
* 499afb9 <bigo@crisidev.org> 2021-01-30 17:42:47 +0000 Do not bind to localhost if we want to reach the internet
* 50594cc <bigo@crisidev.org> 2021-01-30 17:42:20 +0000 Add new crates dependencies
* c92c9ac <bigo@crisidev.org> 2021-01-30 14:44:48 +0000 Save config when removing a peer

### CODE REVIEW

File: src/bin/fireguard.rs [hunk 1]
```diff
@@ -6,7 +6,10 @@ use color_eyre::eyre::Result;
 
 use fireguard::run;
 
-#[tokio::main]
+/// Since reqwest uses a tokio task underneath, I believe we need at
+/// least 2 available threads to ensure we can run both the upgrade task
+/// and the loop waiting for signals asyncronously
+#[tokio::main(worker_threads = 2)]
```
Comment (crisidev) [inline src/bin/fireguard.rs:12]: Yep, this looks to be the issue we ran into for most of the afternoon!

Comment (uovobw) [reply src/bin/fireguard.rs:12]: interesting! i wonder if there is a way of either detecting this based on the number of cpu cores or, even better, if there is a smart configuration for tokio to automatically do this

File: src/github.rs [hunk 2]
```diff
@@ -38,7 +38,7 @@ pub struct Releases {
     pub tarball_url: String,
     #[serde(rename = "zipball_url")]
     pub zipball_url: String,
-    pub body: String,
+    pub body: Option<String>,
```
Comment (uovobw) [inline src/github.rs:41]: this one and the previous ones will need either to make sure we make all fields `Option` (as they have changed with no warning  in the last couple of weeks) or autogenerate this based on some codegen we can run at build time, but then we risk having non-compiling code.... created #35 for this

**Review by uovobw — APPROVED** [review]
good stuff, lgtm, thanks

**Pull request merged**